repo/common/common.go

```diff
@@ -2,26 +2,50 @@ package common
 
 import (
 	"fmt"
+	"regexp"
 	"strings"
 )
 
+// Find dynamic map key names passed  as Parent["foo"] notation
+var bracketsRe = regexp.MustCompile(`\[([^\[\]]*)\]`)
+
+// Normalization for supporting arbitrary dynamic keys with dots:
+// Gateway.PublicGateways["gw.example.com"].UseSubdomains
+// Pinning.RemoteServices["pins.example.org"].Policies.MFS.Enable
+func keyToLookupData(key string) (normalizedKey string, dynamicKeys map[string]string) {
```

Comment:
Putting this here isn't sufficient since we need to deal with things like https://github.com/ipfs/go-ipfs/blob/eea198f0811f50ab14c4f713de07ce93dffec9af/core/commands/config.go#L90 that exist above this level.

We'll either need to export some of this behavior from here, or move the other behavior here and export it.

Reply:
Refactored in 660cc66d4 and added tests in 6a409fba3 (had no docker set up locally, so waiting for CI to run them).

ps. I restored `%s` in error messages because entire error string gets sanitized already upstream, so we want to avoid double escaping.

Reply:
Sharness tests fail on CI but pass locally, I'm suspecting different versions of `jq`, needs another look.

Reply:
I made our CI use the latest `jq` instead of old 1.5, and it is green now.